jena-base/src/main/java/org/apache/jena/atlas/lib/FileOps.java

```diff
@@ -66,9 +66,11 @@ public static void delete(File f, boolean reportExistsAfter) {
     /** Delete all files in a directory */
     public static void clearDirectory(String dir) {
         File d = new File(dir) ;
-        for ( File f : d.listFiles() ) {
-            if ( f.isFile() )
-                delete(f, false) ;
+        if ( d.exists() && d.isDirectory() ) {
```

Comment:
[isDirectory()](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#isDirectory--) actually seems to test for existence by itself, if I read "Return: true if and only if the file denoted by this abstract pathname exists and is a directory; false otherwise" correctly.

Reply:
Thanks! Changed.